splunk-cloudwatch-logs-processor/index.js

```diff
@@ -185,16 +185,58 @@ const CloudWatchToSplunk = (parsed, context, logger, sourcetype, callback) => {
             - Set or remove metadata properties as needed. For descripion of each property, refer to:
             http://docs.splunk.com/Documentation/Splunk/latest/RESTREF/RESTinput#services.2Fcollector */
 
-            const log = {
-                message: item.message,
-                metadata: {
-                    time: item.timestamp ? new Date(item.timestamp).getTime() / 1000 : Date.now(),
-                    host: parsed.logGroup,
-                    source: parsed.logStream,
-                    sourcetype: sourcetype,
-                    //index: 'main',
-                },
-             };
+            let log;
+
+            try {
+                log = JSON.parse(item.message);
+            }
+
+            // Ignore SyntaxError exception if message not in JSON format
+            catch (e) { }
+
+            // Check if message is already in JSON format
+            if (typeof log === "object" && message in log) {
```

Comment:
We think that even if a JSON message has no `message` key it should be pasted as JSON.
```suggestion
```

Reply:
> We think that even if a JSON message has no `message` key it should be pasted as JSON.

@ddriddle , I'll have to disagree with you on this one. The obvious use case is an application that's sending out logs in JSON that has absolutely nothing to do with Splunk metadata JSON. In that case, you'd still want to fall back on the old behavior to add source type, host, etc. and send the log event through to Splunk as the event. Splunk will make the JSON pretty in the index, but only if you put it in the "message" key of the json being sent through. The obvious risk here is what if the application was sending logs in JSON that had nothing to do with Splunk metadata but those logs had a field called "message"?

There are certainly more clever ways to do this as we discussed on the phone a few weeks ago with the Splunk team. For instance, one could add a boolean SSM variable for each service's cloudwatch to splunk variables called something like "logs_in_splunk_json_format". Yeah, that's a bad name, but I think you get the idea. If that was set to true, we'd know to parse the JSON. If it wasn't, we'd fall back on the old behavior. I might suggest that as the next iteration of this code, but it's going to require a bit more work.